unistd/pwd.c

```diff
@@ -24,69 +24,88 @@
 
 struct passwd pwnam;
 char pw_name[NAME_MAX];
+char intstr[16];
 char pw_gecos[128];
 char pw_dir[PATH_MAX];
 char pw_shell[PATH_MAX];
 char pw_passwd[128];
 
 
-struct passwd *getpwnam(const char *name)
+static void readpwEntry(FILE *fp, char *pwentry)
 {
-	FILE *fp;
-	char buf[1024];
+	char c = 0;
+	unsigned int i = 0;
+	c = fgetc(fp);
+	while (c != ':' && c != '\n' && c != EOF) {
+		pwentry[i++] = c;
+		c = fgetc(fp);
+	}
+	pwentry[i] = 0;
```

Comment:
The idea behind not checking it was that user is not supposed to change /etc/passwd file and we can assume its correctness. However, after giving it a second thought you are right. It was changed so that it discards all characters above buffer limit. Thanks!